3ds Max/Max2Babylon/Exporter/BabylonExporter.cs

```diff
@@ -179,6 +179,62 @@ public void FlattenHierarchy(IINode node)
             }
         }
 
+        public void BakeAnimationsFrame(IINode node,BakeAnimationType bakeAnimationType)
+        {
+            if (bakeAnimationType == BakeAnimationType.DoNotBakeAnimation) return;
+
+            IINode hierachyRoot = (node != null) ? node : Loader.Core.RootNode;
+
+#if MAX2020
+            var tobake = Loader.Global.INodeTab.Create();
+#else
+            var tobake = Loader.Global.NodeTab.Create();
+#endif
+            foreach (IINode iNode in hierachyRoot.NodeTree())
+            {
+                tobake.AppendNode(iNode,false,Loader.Core.Time);
+            }
+            if (!hierachyRoot.IsRootNode) tobake.AppendNode(hierachyRoot,false,Loader.Core.Time);
+
+            Loader.Core.SelectNodeTab(tobake,true,false);
+
+            if (bakeAnimationType == BakeAnimationType.BakeAllAnimations)
+            {
+                foreach (IINode n in Tools.ITabToIEnumerable(tobake))
+                {
+                    n.SetUserPropBool("babylonjs_BakeAnimation", true);
+                }
+            }
+
+            ScriptsUtilities.ExecuteMaxScriptCommand(@"
```

Comment:
As I was saying in my previous comment, this pre-pass may not be necessary, since we retrieve each node's transform each frame for our animation.